action/action-util/src/main/java/com/powsybl/action/util/ProportionalScalable.java

```diff
@@ -124,23 +124,28 @@ private void updateIterationPercentages() {
         });
     }
 
-    private double iterativeScale(Network n, double asked, ScalingConvention scalingConvention) {
+    private double iterativeScale(Network n, double asked, ScalingConvention scalingConvention, boolean constantPowerFactor) {
         double done = 0;
         while (Math.abs(asked - done) > EPSILON && notSaturated()) {
             checkIterationPercentages();
-            done += scaleIteration(n, asked - done, scalingConvention);
+            done += scaleIteration(n, asked - done, scalingConvention, constantPowerFactor);
             updateIterationPercentages();
         }
         return done;
     }
 
-    private double scaleIteration(Network n, double asked, ScalingConvention scalingConvention) {
+    private double scaleIteration(Network n, double asked, ScalingConvention scalingConvention, boolean constantPowerFactor) {
         double done = 0;
         for (ScalablePercentage scalablePercentage : scalablePercentageList) {
             Scalable s = scalablePercentage.getScalable();
             double iterationPercentage = scalablePercentage.getIterationPercentage();
             double askedOnScalable = iterationPercentage / 100 * asked;
-            double doneOnScalable = s.scale(n, askedOnScalable, scalingConvention);
+            double doneOnScalable = 0;
+            if (constantPowerFactor && s instanceof LoadScalable) {
```

Comment:
You actually don't even need to check if it is a `LoadScalable`. If you always call `scaleWithConstantPowerFactor` if `constantPowerFactor` is `ŧrue`, it will work!